planner/core/fragment.go

```diff
@@ -125,6 +126,41 @@ func partitionPruning(ctx sessionctx.Context, tbl table.PartitionedTable, conds
 // single physical table means a table without partitions or a single partition in a partition table.
 func (e *mppTaskGenerator) constructMPPTasksImpl(ctx context.Context, ts *PhysicalTableScan) ([]*kv.MPPTask, error) {
 	if ts != nil {
+		// update ranges according to correlated columns in access conditions like in the Open() of TableReaderExecutor
+		hasCorCol := false
+		for _, cond := range ts.AccessCondition {
+			if len(expression.ExtractCorColumns(cond)) > 0 {
+				hasCorCol = true
+				break
+			}
+		}
+		if hasCorCol {
+			access := ts.AccessCondition
+			if ts.Table.IsCommonHandle {
+				pkIdx := tables.FindPrimaryIndex(ts.Table)
+				idxCols, idxColLens := expression.IndexInfo2PrefixCols(ts.Columns, ts.Schema().Columns, pkIdx)
```

Comment:
MPP is not involved in Index query, then what is it for?

Reply:
select pk from t where pk > var,  we can build access ranges according to var using pk.

Reply:
but in this case, pk is not "a common handle"